engine/src/main/java/io/zeebe/engine/state/appliers/ProcessInstanceElementActivatingApplier.java

```diff
@@ -79,7 +79,7 @@ public void applyState(final long elementInstanceKey, final ProcessInstanceRecor
         && MigratedStreamProcessors.isMigrated(currentElementType)) {
       // update the loop counter of the multi-instance body (starting by 1)
       elementInstanceState.updateInstance(
-          value.getFlowScopeKey(), ElementInstance::incrementMultiInstanceLoopCounter);
+          flowScopeInstance, ElementInstance::incrementMultiInstanceLoopCounter);
```

Comment:
(optional) Instead of introducing a new method to update the element instance, we could use the existing one. 

Having multiple/too many ways of doing one thing have some downsides: more complex, more things to test, confusing because of different styles. Expect there is a good reason :) 

```
flowScopeInstance.incrementMultiInstanceLoopCounter();
elementInstanceState.updateInstance(flowScopeInstance);
```

Reply:
I see, interesting point. I thought about making an argument for convenience, but this method is anyways different from all others that can be mentioned here because it mutates the provided parameter. I'll change it. 👍